gossip/src/cluster_info.rs

```diff
@@ -1066,7 +1066,13 @@ impl ClusterInfo {
         }
     }
 
-    fn find_vote_index_to_evict(&self, should_evict_vote: impl Fn(&Vote) -> bool) -> u8 {
+    /// If there are less than `MAX_LOCKOUT_HISTORY` votes present, returns the next index
+    /// without a vote. If there are `MAX_LOCKOUT_HISTORY` votes:
+    /// - Finds the oldest wallclock vote with slot < `new_vote_slot` and returns its index
```

Comment:
why do we need to check for  `slot < new_vote_slot`?
Can't we just pick the one with oldest wallclock regardless of slot?

and, what happens if we switch fork and vote on a slot smaller than our last voted slot?

Reply:
this is a good point, we will never submit a vote for a slot smaller than any previous vote. Given this I believe the correct thing to do is panic if we observe a vote in gossip for a slot > than the vote we are trying to push - as it would indicate that our local state is messed up after a restart.

I've updated it as such and that way we can just pick the oldest wallclock vote.

Reply:
> we will never submit a vote for a slot smaller than any previous vote

So you can never switch to fork with last slot smaller the last slot you voted on?
Why is that?

Reply:
oh right, obviously lol
totally forgot how lockouts work 😅